tests/unit/amo/components/TestCollection.js

```diff
@@ -0,0 +1,445 @@
+
+    sinon.assert.notCalled(fakeDispatch);
+  });
+
+  it('does not dispatch any action when loading collection', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'some-user';
+
+    store.dispatch(fetchCollection({
+      errorHandlerId: errorHandler.id,
+      slug,
+      user,
+    }));
+
+    fakeDispatch.reset();
+    renderComponent({ store });
+
+    sinon.assert.notCalled(fakeDispatch);
+  });
+
+  it('does not dispatch any action when loading collection page', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'some-user';
+
+    store.dispatch(fetchCollectionPage({
+      errorHandlerId: errorHandler.id,
+      page: 123,
+      slug,
+      user,
+    }));
+
+    fakeDispatch.reset();
+    renderComponent({ store });
+
+    sinon.assert.notCalled(fakeDispatch);
+  });
+
+  it('does not dispatch any action when there is an error', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    const wrapper = renderComponent({ store });
+
+    const id = 'error-handler-id';
+    const error = createApiError({
+      response: { status: 500 },
+      apiURL: 'https://some/api/endpoint',
+      jsonResponse: { message: 'Nope.' },
+    });
+    store.dispatch(setError({ id, error }));
+    const capturedError = store.getState().errors[id];
+    expect(capturedError).toBeTruthy();
+
+    const errorHandler = new ErrorHandler({
+      id, dispatch: sinon.stub(), capturedError,
+    });
+
+    fakeDispatch.reset();
+    wrapper.setProps({ errorHandler });
+
+    sinon.assert.notCalled(fakeDispatch);
+  });
+
+  it('dispatches fetchCollection when location pathname has changed', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    // We need a collection for this test case.
+    store.dispatch(loadCollection({
+      addons: createFakeCollectionAddons(),
+      detail: createFakeCollectionDetail(),
+    }));
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'some-user';
+    const page = 123;
+
+    const location = {
+      pathname: `/collections/${user}/${slug}/`,
+      query: { page },
+    };
+
+    const newSlug = 'other-collection';
+    const newLocation = {
+      ...location,
+      pathname: `/collections/${user}/${newSlug}/`,
+    };
+
+    const wrapper = renderComponent({
+      errorHandler,
+      location,
+      params: { slug, user },
+      store,
+    });
+    fakeDispatch.reset();
+
+    // This will trigger the componentWillReceiveProps() method.
+    wrapper.setProps({
```

Comment:
Can we dispatch on the `store` we pass to `renderComponent` here to do this instead? It might be a bit nicer.

Reply:
`location` and `params` are not handled by redux.

Reply:
> location and params are not handled by redux.

I'm looking forward to https://github.com/mozilla/addons-frontend/issues/181 for that